### DISCUSSION TITLE
v0.8.2

### DISCUSSION THREAD

**Discussion by laipz8200** [langgenius/dify#8354] [Releases]
## ✨ What’s New in v0.8.2? ✨

Hey everyone, we've got some fresh updates in version 0.8.2 that are all about making your experience smoother and more efficient. Here's the lowdown:

### 🚀 New Features

- **O1-Series Models Support**: We've added support for the O1-series models in the Agent App, specifically for ReACT workflows. This enhancement by @takatost in #8350 opens up new possibilities for those leveraging ReACT in their applications.

- **Service API Workflow Logs**: Thanks to @fanlia in #8323, you can now access detailed workflow logs via the service API. This is a big win for those of you who need to keep a close eye on what's happening under the hood.

### 🛠️ Bug Fixes

- **Score Threshold Fix**: Addressed an issue where the score threshold was set to none, ensuring more reliable performance metrics by @JohnJyong in #8342.

- **Workflow Var-Selector Update**: Fixed a bug where the var-selector wasn't updating when edges changed in workflows. Kudos to @HiChen404 for this fix in #8259.

- **Copy and Paste Shortcut**: Resolved an issue with the copy and paste shortcut in the textarea of the workflow run panel, making text manipulation a breeze again. Thanks to @YIXIAO0 for the fix in #8345.

That's it for this release! Keep those feedback and contributions coming, and let's keep making this platform even better together.

---

### Docker compose deployments

> [!WARNING]
> The `docker-compose.yaml` has been refactored. If you've made any changes to the file, make sure to check out the "Upgrade to new docker compose deployment" section above for usage and migration tips.

1. Back up your customized docker-compose YAML file (optional)

   ```bash
   cd docker
   cp docker-compose.yaml docker-compose.yaml.$(date +%s).bak
   ```

2. Get the latest code from the main branch

   ```bash
   git checkout main
   git pull origin main
   ```

3. Stop the service，Command, please execute in the docker directory

   ```bash
   docker compose down
   ```

4. Back up data

   ```bash
   tar -cvf volumes-$(date +%s).tgz volumes
   ```

5. Upgrade services

   ```bash
   docker compose up -d
   ```

### Source Code deployments

1. Stop API server, Worker and Web frontend Server.

2. Get the latest code from the main branch:

   ```bash
   git checkout main
   git pull origin main
   ```

3. Update Python dependencies:

   ```bash
   cd api
   poetry install
   ```

7. Then, let's run the migration script:

   ```bash
   poetry shell
   flask db upgrade
   ```

8. Finally, run API server, Worker and Web frontend Server again.

---

## What's Changed
* fixed score threshold is none by @JohnJyong in https://github.com/langgenius/dify/pull/8342
* fix(workflow): fix var-selector not update when edges change by @HiChen404 in https://github.com/langgenius/dify/pull/8259
* Feature/service api workflow logs by @fanlia in https://github.com/langgenius/dify/pull/8323
* fix: copy and paste shortcut in the textarea of the workflow run panel by @YIXIAO0 in https://github.com/langgenius/dify/pull/8345
* feat: add o1-series models support in Agent App (ReACT only) by @takatost in https://github.com/langgenius/dify/pull/8350
* feat: add langfuse sentry ignore error by @ZhouhaoJiang in https://github.com/langgenius/dify/pull/8353
* chore: update version to 0.8.2 in packaging and docker-compose files by @laipz8200 in https://github.com/langgenius/dify/pull/8352

## New Contributors
* @fanlia made their first contribution in https://github.com/langgenius/dify/pull/8323

**Full Changelog**: https://github.com/langgenius/dify/compare/0.8.1...0.8.2

<hr /><em>This discussion was created from the release <a href='https://github.com/langgenius/dify/releases/tag/0.8.2'>v0.8.2</a>.</em>

**Comment by Habib786741** [comment]
Hi

**Comment by Ima-Beast** [comment]
Hello，i update to version 0.8.2 via the source code still does not support multiple branches in the workflow.

**Reply by laipz8200** [reply]
Please take a moment to review our documentation, or utilize our cloud services to check your operations. If you continue to experience any issues, feel free to describe your problem in detail and submit an issue report. We're here to help!
